fdbserver/TLogServer.actor.cpp

```diff
@@ -2296,6 +2297,12 @@ ACTOR Future<Void> commitQueue(TLogData* self) {
 		}
 
 		loop {
+			// Insert enough of a delay to allow this tlog to be stopped and a new one registered
+			// before the commit is issued. These are the conditions which trigger a missingFinalCommit.
+			if (g_network->isSimulated() && !g_simulator->speedUpSimulation && BUGGIFY) {
```

Comment:
Is this the right BUGGIFY invocation? Or should this happen less frequently, e.g. BUGGIFY_WITH_PROB(0.0001)?

Reply:
```suggestion
			if (BUGGIFY && !g_simulator->speedUpSimulation) {
```

I do see your point though about this maybe being a little excessive if a tlog always has a 25% chance of waiting 1s before a commit.  I'd probably tune the probability for BUGGIFY_WITH_PROB such that you'll see this in at least 1 in 10k runs, which means often enough that someone would notice if they broke it.

Alternatively, if you want to specifically target the stop and start case, you can just tack on some extra conditions?  I think the two conditions below give you the check for TLog is stopped, and TLog is newly recruited, and if either of those work to still trigger the bug, then you could keep a much higher buggify rate with likely little penalty.

Reply:
I went with buggify with probability 1/10,000. The stopped condition is interesting, but the bug I found wouldn't have been caught with the second `logData->queueCommittedVersion.get() == std::numeric_limits<Version>::max()`, since it was initialized with a queue committed version of one less than its initial version.